L1Trigger/TrackerTFP/src/SeedFilter.cc

```diff
@@ -0,0 +1,242 @@
+#include "L1Trigger/TrackerTFP/interface/SeedFilter.h"
+
+#include <numeric>
+#include <algorithm>
+#include <iterator>
+#include <deque>
+#include <vector>
+#include <set>
+#include <utility>
+#include <cmath>
+
+using namespace std;
+using namespace edm;
+using namespace trackerDTC;
+
+namespace trackerTFP {
+
+  SeedFilter::SeedFilter(const ParameterSet& iConfig, const Setup* setup, const DataFormats* dataFormats, const LayerEncoding* layerEncoding, int region) :
+    enableTruncation_(iConfig.getParameter<bool>("EnableTruncation")),
+    setup_(setup),
+    dataFormats_(dataFormats),
+    layerEncoding_(layerEncoding),
+    region_(region),
+    cot_(dataFormats_->format(Variable::cot, Process::sf)),
+    zT_(dataFormats_->format(Variable::zT, Process::sf)),
+    z_(dataFormats_->format(Variable::z, Process::sf)),
+    input_(dataFormats_->numChannel(Process::mht)) {}
+
+  void SeedFilter::consume(const TTDTC::Streams& streams) {
+    auto valid = [](int& sum, const TTDTC::Frame& frame){ return sum += (frame.first.isNonnull() ? 1 : 0); };
```

Comment:
Add comment "Read in & organise input data (filling array input_)"

Reply:
done